darts/models/theta.py

```diff
@@ -126,3 +127,217 @@ def predict(self, n: int) -> 'TimeSeries':
+        self.coef = 1
+        self.mean = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = seasonality_period
+        self.model_mode = model_mode
+        self.season_mode = season_mode
+        self.trend_mode = trend_mode
+        self.wses = 0 if self.theta == 0 else (1 / theta)
+        self.wdrift = 1 - self.wses
+        self.fitted_values = None
+
+        # Remark on the values of the theta parameter:
+        # - if theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        # - if theta = 0, then the theta method restricts to a simple `trend_mode` regression.
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        self.mean = ts.mean()
+        new_ts = ts / self.mean
+
+        # Check for statistical significance of user-defined season period
+        # or infers season_period from the TimeSeries itself.
+        if self.season_period is None:
+            max_lag = len(ts) // 2
+            self.is_seasonal, self.season_period = check_seasonality(ts, self.season_period, max_lag=max_lag)
+            logger.info('Theta model inferred seasonality of training series: {}'.format(self.season_period))
+        else:
+            # force the user-defined seasonality to be considered as a true seasonal period.
+            self.is_seasonal = self.season_period > 1
+
+        # Store and remove seasonality effect if there is any.
+        if self.is_seasonal:
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period, model=self.season_mode)
+            new_ts = remove_seasonality(new_ts, self.season_period, model=self.season_mode)
+
+        if (new_ts <= 0).values.any():
+            self.model_mode = 'additive'
+            self.trend_mode = 'linear'
+            logger.warn("Time series has negative values. Fallback to additive and linear model")
+
+        # Drift part of the decomposition
+        if self.trend_mode == 'linear':
+            linreg = new_ts.values()
+        elif self.trend_mode == 'exponential':
+            linreg = np.log(new_ts.values())
+        else:
+            self.trend_mode = 'linear'
+            linreg = new_ts.values()
+            logger.warn("Unknown value for trend. Fallback to linear.")
+        self.drift = np.poly1d(np.polyfit(np.arange(self.length), linreg, 1))
+        theta0_in = self.drift(np.arange(self.length))
+        if self.trend_mode == 'exponential':
+            theta0_in = np.exp(theta0_in)
+
+        if self.model_mode == 'additive':
+            theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+        elif self.model_mode == 'multiplicative' and (theta0_in > 0).all():
+            theta_t = (new_ts.values() ** self.theta) * (theta0_in ** (1 - self.theta))
+        else:
```

Comment:
For a first draft, I blindly replicated the step of the original alorithm. So I will change it. 
But we still need to check if all values are positive if 'multiplicative' models are chosen or else it will fail. 
I can either raise an error, or simply keep the fallback as it is.

Reply:
There should be a more optimized implementation, like our theta. But we need to do the math to find it